manager/manager.go

```diff
@@ -794,7 +842,30 @@ func (self *manager) GetFsInfo(label string) ([]v2.FsInfo, error) {
 
 func (m *manager) GetMachineInfo() (*info.MachineInfo, error) {
 	// Copy and return the MachineInfo.
-	return &m.machineInfo, nil
+	m.machineInfoLock.RLock()
```

Comment:
I would prefer just reconstructing machineInfo each time instead of watching for disk mount changes.  We can assume that inHostNamespace doesnt change, so we can avoid repeatedly calling `os.Stat("/rootfs/proc")`.  But this watching for events is more complexity than it is worth.

Reply:
I have some questions:
1.If  reconstructing machineInfo each time , what's the interval of reconstructing machineInfo? It cann't be too soon, or it will waste cpu and memory.  What about  30s or 1 minute? 
2. I would move  inHostNamespace  into the struct `manager` , is that reasonable? How do you think?

Reply:
I looked a little closer and this is a expensive operation than I thought.  Can we only update the Filesystems field in machineInfo, and keep the rest?